core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -80,8 +86,59 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   //periodically emit metrics (don't need to do this for each message!)
   context.system.scheduler.schedule(30.seconds, 10.seconds, self, EmitMetrics)
 
+  // Key is `kind,memory,time`, value is the number of cold Start in minute
+  var coldStartCount = immutable.Map.empty[String, Int]
```

Comment:
It seems this map can be used in multiple threads, should we use `concurrent.map` instead?

Reply:
Oh, really, i think `coldStartCount` can only be accessed in single thread.

Reply:
immutable.Map should be fine if it is only modified during actor message processing

Reply:
oh ok I thought it was being accessed by a scheduler and actor thread at the same time.
But it was not.